python/HeaderService/hslib_salobj.py

```diff
@@ -671,11 +683,25 @@ async def announce(self, imageName):
 
         await self.evt_largeFileObjectAvailable.set_write(**kw)
         self.log.info(f"Sent {self.config.hs_name} largeFileObjectAvailable: {kw}")
+        self.completed_OK[imageName] = True
+
+    async def upload_to_s3(self, fileobj, key, imageName, nt=2):
+        s3upload = False
+        k = 1
+        while k <= nt and s3upload is False:
+            try:
+                await self.s3bucket.upload(fileobj=fileobj, key=key)
+                s3upload = True
+            except Exception as e:
+                self.log.error(f"Failed s3bucket.upload attempt # {k}/{nt} for: {imageName}")
+                self.log.exception(str(e))
+            k += 1
```

Comment:
Maybe do a `time.sleep(0.1)` before retrying?  (Well, ideally make the delay a module constant or configuration value rather than hard-coding it.  You could also make the number of retries a constant or configuration value.)

Reply:
It's not needed. The internal `s3bucket.upload` has a long wait built in.